math/ncr_modulo_p.cpp

```diff
@@ -0,0 +1,148 @@
+/**
+ * @file
+ * @brief This program aims at calculating nCr modulo p. To know more about it,
+ * visit https://cp-algorithms.com/combinatorics/binomial-coefficients.html
+ * @details nCr is defined as n! / (r! * (n-r)!) where n! represents factorial
+ * of n. In many cases, the value of nCr is too large to fit in a 64 bit
+ * integer. Hence, in competitive programming, there are many problems or
+ * subproblems to compute nCr modulo p where p is a given number.
+ * @author [Kaustubh Damania](https://github.com/KaustubhDamania)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for io operations
+#include <vector>    /// for std::vector
+
+/**
+ * @namespace math
+ * @brief Mathematical algorithms
+ */
+namespace math {
+/**
+ * @namespace ncr_modulo_p
+ * @brief Functions for nCr modulo p implementation.
+ */
+namespace ncr_modulo_p {
+/**
+ * @brief Class which contains all methods required for calculating nCr mod p
+ */
+class NCRModuloP {
+ private:
+    std::vector<uint64_t> fac{};
+    uint64_t p = 0;
```

Comment:
Add a brief description of what the variable does.
```suggestion
    std::vector<uint64_t> fac{};  ///< brief description
    uint64_t p = 0;               ///< brief description
```

Reply:
Done! 👍